plugins/optimization-detective/site-health.php

```diff
@@ -0,0 +1,293 @@
+}
+
+/**
+ * Tests availability of the Optimization Detective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function od_test_rest_api_availability(): array {
+	$response       = od_get_rest_api_health_check_response( false );
+	$result         = od_compose_site_health_result( $response );
+	$is_unavailable = 'good' !== $result['status'];
+	update_option(
+		'od_rest_api_unavailable',
+		$is_unavailable ? '1' : '0',
+		true // Intentionally autoloaded since used on every frontend request.
+	);
+	return $result;
+}
+
+/**
+ * Checks whether the Optimization Detective REST API endpoint is unavailable.
+ *
+ * This merely checks the database option what was previously computed in the Site Health test as done in {@see od_test_rest_api_availability()}.
+ * This is to avoid checking for REST API availability during a frontend request. Note that when the plugin is first
+ * installed, the 'od_rest_api_unavailable' option will not be in the database, as the check has not been performed
+ * yet. Once Site Health's weekly check happens or when a user accesses the admin so that the admin_init action fires,
+ * then at this point the check will be performed at {@see od_maybe_run_rest_api_health_check()}. In practice, this will
+ * happen immediately after the user activates a plugin since the user is redirected back to the plugin list table in
+ * the admin. The reason for storing the negative unavailable state as opposed to the positive available state is that
+ * when an option does not exist then `get_option()` returns `false` which is the same falsy value as the stored `'0'`.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @return bool Whether unavailable.
+ */
+function od_is_rest_api_unavailable(): bool {
+	return 1 === (int) get_option( 'od_rest_api_unavailable', '0' );
+}
+
+/**
+ * Tests availability of the Optimization Detective REST API endpoint.
+ *
+ * @since n.e.x.t
+ * @access private
+ *
+ * @param array<string, mixed>|WP_Error $response REST API response.
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function od_compose_site_health_result( $response ): array {
+	$common_description_html = '<p>' . wp_kses(
+		sprintf(
+			/* translators: %s is the REST API endpoint */
+			__( 'To collect URL Metrics from visitors the REST API must be available to unauthenticated users. Specifically, visitors must be able to perform a <code>POST</code> request to the <code>%s</code> endpoint.', 'optimization-detective' ),
+			'/' . OD_REST_API_NAMESPACE . OD_URL_METRICS_ROUTE
+		),
+		array( 'code' => array() )
+	) . '</p>';
+
+	$result = array(
+		'label'       => __( 'Optimization Detective\'s REST API endpoint is available', 'optimization-detective' ),
```

Comment:
WP Coding Standards encourage to not escape `'` and `"`, but instead use the respective other to wrap the string, so I think we should do that here if we want to keep the string as is.

That said, I think it's unnecessary. Using "The Optimization Detective REST API endpoint" like it is already used below in line 139 sounds better to me.
```suggestion
		'label'       => __( 'The Optimization Detective REST API endpoint is available', 'optimization-detective' ),
```